geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/DestroyAsyncEventQueueCommand.java

```diff
@@ -0,0 +1,99 @@
+import org.apache.geode.management.internal.cli.result.TabularResultData;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+public class DestroyAsyncEventQueueCommand implements GfshCommand {
+  public static final String DESTROY_ASYNC_EVENT_QUEUE = "destroy async-event-queue";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__HELP = "destroy an Async Event Queue";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__ID = "id";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__ID__HELP =
+      "ID of the queue to be destroyed.";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__GROUP__HELP =
+      "Group(s) of members on which to destroy the async event queue.";
+
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__AEQ_0_NOT_FOUND =
+      "Async event queue \"%s\" not found";
+  public static final String DESTROY_ASYNC_EVENT_QUEUE__AEQ_0_DESTROYED =
+      "Async event queue \"%s\" destroyed";
+
+  @CliCommand(value = DESTROY_ASYNC_EVENT_QUEUE, help = DESTROY_ASYNC_EVENT_QUEUE__HELP)
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.MANAGE)
+  public Result destroyAsyncEventQueue(
+      @CliOption(key = DESTROY_ASYNC_EVENT_QUEUE__ID, mandatory = true,
+          help = DESTROY_ASYNC_EVENT_QUEUE__ID__HELP) String aeqId,
+      @CliOption(key = {CliStrings.GROUP, CliStrings.GROUPS},
+          optionContext = ConverterHint.MEMBERGROUP,
+          help = DESTROY_ASYNC_EVENT_QUEUE__GROUP__HELP) String[] onGroups,
+      @CliOption(key = IFEXISTS, help = IFEXISTS_HELP, specifiedDefaultValue = "true",
+          unspecifiedDefaultValue = "false") boolean ifExists) {
+    DestroyAsyncEventQueueFunctionArgs asyncEventQueueDestoryFunctionArgs =
+        new DestroyAsyncEventQueueFunctionArgs(aeqId, ifExists);
+
+    Set<DistributedMember> members = getMembers(onGroups, null);
+
+    List<CliFunctionResult> functionResults =
+        execute(new DestroyAsyncEventQueueFunction(), asyncEventQueueDestoryFunctionArgs, members);
+
+    TabularResultData tabularResultData = ResultBuilder.createTabularResultData();
+    boolean errorOccurred = false;
+    for (CliFunctionResult functionResult : functionResults) {
+      LogService.getLogger().info("FunctionResult = '" + functionResult + "'");
+      tabularResultData.accumulate("Member", functionResult.getMemberIdOrName());
+      if (functionResult.isSuccessful()) {
+        tabularResultData.accumulate("Status", functionResult.getMessage());
+      } else {
+        // if result has exception, it will be logged by the server before throwing it.
+        // so we don't need to log it here anymore.
+        tabularResultData.accumulate("Status", "ERROR: " + functionResult.getErrorMessage());
+        errorOccurred = true;
+      }
+    }
+    tabularResultData.setStatus(errorOccurred ? Result.Status.ERROR : Result.Status.OK);
```

Comment:
You will need to update the cluster configuration as well. Have your function return an xmlEntity, and call a clusterConfigService.deleteXmlEntity(). Use DestroyRegionCommand as an example.

Reply:
Fixed.